source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc

```diff
@@ -88,6 +102,18 @@ void GrpcClientImpl::onFailure(Grpc::Status::GrpcStatus status, const std::strin
   ENVOY_LOG(trace, "CheckRequest call failed with status: {}",
             Grpc::Utility::grpcStatusToString(status));
   ASSERT(status != Grpc::Status::WellKnownGrpcStatus::Ok);
+  respondFailure();
+}
+
+void GrpcClientImpl::onTimeout() {
+  ENVOY_LOG(trace, "CheckRequest timed-out");
+  ASSERT(request_ != nullptr);
```

Comment:
Do you need to disable the timer on a successful response? And in `onFailure`?

Reply:
good catch! fixing and adding tests